README.md

```diff
@@ -19,52 +19,191 @@ Works well with [redux-optimist-promise](https://github.com/mathieudutour/redux-
 
 ## Usage
 
+*actions.js*
+```js
+// We define our action types here so that they can easily be used by different ducks
+export const FETCH_TODOS = 'todo.FETCH_TODOS';
+export const ADD_TODO = 'todo.ADD_TODO';
+export const LOGOUT = 'auth.LOGOUT';
+```
+
+*ducks/todos.js*
 ```js
 import Duck from 'reduck'
 
 import {
+  ADD_TODO,
   FETCH_TODOS,
   LOGOUT,
-} from '../actions';
+} from '../actions'
 
+// define an initial state to use in the Duck's initialization
 const initialState = {
   items: [],
-  ready: false,
+  ready: false
 };
 
 const duck = new Duck('todo', initialState);
 
+export const addTodo = duck.defineAction(ADD_TODO, {
+ creator(newTodoItem) {
+  return {
+   payload: { newTodoItem }
+  };
+ },
+ reducer(state, { payload }) {
+  return {
+   ...state,
+   items: (state.items || []).concat(payload.newTodoItem)
+ }
+})
+
+// action defined in another duck but this duck still want to react to it
+duck.addReducerCase(LOGOUT, {
+ reducer() {
+  return initialState;
+ }
+})
+
+export default duck.reducer
+
+```
+
+*ducks/auth.js*
+```js
+import Duck from 'reduck'
+import { LOGOUT } from '../actions'
+
+const initialState = {
+ user: {},
+ authStatus: 'Unknown'
+}
+
+const duck = new Duck('auth', initialState);
+
+duck.defineAction(LOGOUT, {
+ creator() {
+  return {} // Data is not necessary in this case
+ },
+ reducer(state) {
+  return initialState
+ }
+})
+
+export default duck.reducer;
+
+```
+
+*reducer.js*
+```js
+import { combineReducers } from 'redux'
+
+import auth from './ducks/auth'
+import todos from './ducks/todos'
+
+export default combineReducers({
+ auth, todos,
+})
+```
+
+### API
+
+`constructor(duckName, initialState)`
+- `duckName`: Given name of the duck
+- `initialState`: the state that the duck will be initialized with
+- returns a 'duck' object that provides the following methods:
+  
+**Methods**
+
+`defineAction(actionType: String, reducerCases: Object)`
+- `actionType` is the type of the action. Can be whatever you choose as long as it follows the format: `<duck-name>.<some-action-name>`.
+This helps with tracking defined actions in each duck.
+
+- `reducerCases` consist of:
+  * `creator(actionArgs)`
+  The creator accepts any arguments related to the action being performed and then returns the object that will be passed to the action's reducers. The object consists of mainly `payload` (the payload that will be handled by the reducers) and `meta` (data that will be used by middleware). 
+  _Note:_The creator **must** be present in the `defineAction` method.
```

Comment:
Yeah I was thinking about that when writing it. 
I don't mind having `creator() { return {} }` that much because it does remind to anyone looking at the code that this action doesn't require any payload (and it's not that much code)